pkgs/esp-idf/default.nix

```diff
@@ -1,6 +1,6 @@
 # When updating to a newer version, check if the version of `esp32-toolchain-bin.nix` also needs to be updated.
-{ rev ? "v4.3.1"
-, sha256 ? "sha256-+SMdnIBSCdHy+MDbInl/aXJuXOf9seJbQ3u4mN+qFP4="
+{ rev ? "v4.4.1"
```

Comment:
Since you updated this, please also update the ESP32 and ESP32-C3 toolchain versions to `2021r2-patch3` - they all need to be the right version for the IDF version, otherwise the IDF complains.